### PR TITLE
Get rid of libmacouflage library

### PR DESCRIPTION
**Pull request by dteplits**
Remove unnecessary usage of libmacouflage library to save
~25MB of memory per VM
libmacouflage library was used to generate MAC address for Pod NIC during 
preparations for the creation of VMI.
In order to spare the memory cost of using a vendor library, 
Pod NIC MAC address will be populated with bridge initial MAC address instead of generating
a new one.
Bridge MAC initial address is being overridden by the first link so it's safe to reuse it.

Signed-off-by: Diana Teplits <dteplits@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```
Memory overhead reduced by ~25MB per VMI
```

### CODE REVIEW

**Comment by dteplits** [comment]
\test pull-kubevirt-e2e-k8s-1.22-sig-network

**Comment by orelmisan** [comment]
Hi @dteplits, there is another reference to `libmacouflage` in `go.mod`.

**Review by RamLavi — APPROVED** [review]
/lgtm

File: pkg/network/infraconfigurators/bridge.go [hunk 1]
```diff
@@ -197,6 +193,17 @@ func (b *BridgePodNetworkConfigurator) createBridge() error {
...
+	//replace podNic's mac with bridge's mac before linking podIface as master on bridge
+	brMac, err := b.handler.GetMac(b.bridgeInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to access bridge")
+		return err
+	}
+
+	if err := b.handler.SetMac(b.podNicLink.Attrs().Name, brMac.String()); err != nil {
+		return err
+	}
```
Comment (EdDev) [inline pkg/network/infraconfigurators/bridge.go:205]: This is equivalent to directly doing:
```
link, err := h.LinkByName(ifaceName)
if err != nil {
	return nil, err
}
return netlink.LinkSetHardwareAddr(link, link.Attrs().HardwareAddr)
```

So why do we need those helper functions? They seem to complicate things.

There is probably an issue with using `nelink` directly, but that is true for both versions.

File: pkg/network/infraconfigurators/bridge.go [hunk 2]
```diff
@@ -197,6 +193,17 @@ func (b *BridgePodNetworkConfigurator) createBridge() error {
 		return err
 	}
 
+	//replace podNic's mac with bridge's mac before linking podIface as master on bridge
```
Comment (EdDev) [inline pkg/network/infraconfigurators/bridge.go:196]: Well, I see this written just  below using code (which is not really complex). I do not see additional information which is beyond the code (e.g. reasons or some context to the outside).

If you still think it is important, I prefer to see comments on the function description.

**Review by EdDev — CHANGES_REQUESTED** [review]
Thank you.
There are some comments inline.

Regarding the commit messages:
- Please add a few words of what existed now and why that library was used.
- Why it is not really needed (e.g. using the bridge mac is good enough).
- Structure the subject like to have a topic. e.g. `network: Use the bridge mac on the pod iface`
  I think that the drop of the library is a subsequence of what this logical change does.
  The reasoning is that we want to simplify the logic and drop unneeded dependencies, which in turn caused high memory consumption.
  
BTW, I know this is not in the scope of this change, but I find the unit tests unreadable. I have hard time trusting them, but I recognize it is out of this PR scope.

File: pkg/network/infraconfigurators/bridge.go (hunk 2 above)
Comment (dteplits) [reply pkg/network/infraconfigurators/bridge.go]: fixed

**Review by EdDev — CHANGES_REQUESTED** [review]
Some small comments inline.
There are also some typos and missing dot/s (`.`) at the end of sentences in the commit messages.

File: pkg/network/infraconfigurators/bridge_test.go [hunk 3]
```diff
@@ -202,7 +201,6 @@ var _ = Describe("Bridge infrastructure configurator", func() {
 		BeforeEach(func() {
 			iface = v1.DefaultBridgeNetworkInterface()
 			vmi = newVMIWithBridgeInterface("default", "vm1")
-			mac, _ = net.ParseMAC(macStr)
```
[the remark this answers is not in the document]
Comment (dteplits) [reply pkg/network/infraconfigurators/bridge_test.go:205]: macStr removed

File: pkg/network/infraconfigurators/bridge.go [hunk 4]
```diff
@@ -197,6 +193,17 @@ func (b *BridgePodNetworkConfigurator) createBridge() error {
...
 	}
 
+	brLink, err := b.handler.LinkByName(b.bridgeInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to access bridge")
+		return err
+	}
+
+	if err := b.handler.LinkSetHardwareAddr(b.podNicLink, brLink.Attrs().HardwareAddr); err != nil {
+		log.Log.Reason(err).Errorf("failed to set a bridge mac in the renamed pod link")
```
[the remark this answers is not in the document]
Comment (dteplits) [reply pkg/network/infraconfigurators/bridge.go]: fixed

**Comment by EdDev** [comment]
@dteplits , something went wrong with your rebase. This PR now includes commits which are not suppose to be here.

**Review by EdDev — APPROVED** [review]
Thank you!

**Comment by AlonaKaplan** [comment]
/approve

Thanks! Well done:)

**Pull request merged**